magicblock-committor-service/src/persist/commit_persister.rs

```diff
@@ -1,258 +1,574 @@
 use std::{
     path::Path,
-    sync::atomic::{AtomicU64, Ordering},
+    sync::{Arc, Mutex},
 };
 
-use magicblock_committor_program::Changeset;
-use solana_sdk::{hash::Hash, pubkey::Pubkey};
+use magicblock_program::magic_scheduled_base_intent::ScheduledBaseIntent;
+use solana_sdk::pubkey::Pubkey;
 
 use super::{
-    db::{BundleSignatureRow, CommitStatusRow},
-    error::{CommitPersistError, CommitPersistResult},
-    utils::now,
-    CommitStatus, CommitType, CommittorDb,
+    db::CommitStatusRow, error::CommitPersistResult, utils::now, CommitStatus,
+    CommitStrategy, CommitType, CommittsDb, MessageSignatures,
 };
 
-pub struct CommitPersister {
-    db: CommittorDb,
-    request_id_counter: AtomicU64,
+const POISONED_MUTEX_MSG: &str = "Commitor Persister lock poisoned";
+
+/// Records lifespan pf BaseIntent
+pub trait IntentPersister: Send + Sync + Clone + 'static {
+    /// Starts persisting BaseIntent
+    fn start_base_intents(
+        &self,
+        base_intent: &[ScheduledBaseIntent],
+    ) -> CommitPersistResult<()>;
+    fn start_base_intent(
+        &self,
+        base_intent: &ScheduledBaseIntent,
+    ) -> CommitPersistResult<()>;
+    fn set_commit_id(
+        &self,
+        message_id: u64,
+        pubkey: &Pubkey,
+        commit_id: u64,
+    ) -> CommitPersistResult<()>;
+    fn set_commit_strategy(
+        &self,
+        commit_id: u64,
+        pubkey: &Pubkey,
+        value: CommitStrategy,
+    ) -> CommitPersistResult<()>;
+    fn update_status_by_message(
+        &self,
+        message_id: u64,
+        pubkey: &Pubkey,
+        status: CommitStatus,
+    ) -> CommitPersistResult<()>;
+    fn update_status_by_commit(
+        &self,
+        commit_id: u64,
+        pubkey: &Pubkey,
+        status: CommitStatus,
+    ) -> CommitPersistResult<()>;
+    fn get_commit_statuses_by_message(
+        &self,
+        message_id: u64,
+    ) -> CommitPersistResult<Vec<CommitStatusRow>>;
+    fn get_commit_status_by_message(
+        &self,
+        message_id: u64,
+        pubkey: &Pubkey,
+    ) -> CommitPersistResult<Option<CommitStatusRow>>;
+    fn get_signatures_by_commit(
+        &self,
+        commit_id: u64,
+        pubkey: &Pubkey,
+    ) -> CommitPersistResult<Option<MessageSignatures>>;
+    // fn finalize_l1_message(&self blockhash: Hash) -> CommitPersistResult<()>;
 }
 
-impl CommitPersister {
+#[derive(Clone)]
+pub struct IntentPersisterImpl {
+    // DB that tracks lifespan of Commit intents
+    commits_db: Arc<Mutex<CommittsDb>>,
+    // TODO: add something like
+    // actions_db: Arc<Mutex<ActionsDb>>
```

Comment:
Not needed for now? At least add your github handle to the TODO.
I'm not a huge fan of adding implementation reminders to the code. Maybe make a TODO file locally for yourself pointing to the location in the files where to add the features later instead?